File: midterm lab 3/java/MP10.java
import java.io.*;
import java.util.*;

// MP10 - Detect duplicate records
public class MP10 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        try {
            System.out.print("Enter CSV file path: ");
            String filePath = scanner.nextLine();

            BufferedReader br = new BufferedReader(new FileReader(filePath));

            HashSet<String> unique = new HashSet<>();
            HashSet<String> duplicates = new HashSet<>();

            String line;

            // Read each row
            while ((line = br.readLine()) != null) {
                if (!unique.add(line)) {
                    duplicates.add(line); // already exists
                }
            }

            br.close();

            System.out.println("\nDuplicate Records:");
            if (duplicates.isEmpty()) {
                System.out.println("No duplicates found.");
            } else {
                for (String dup : duplicates) {
                    System.out.println(dup);
                }
            }

        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

File: midterm lab 3/javascript/MP10.js
// MP10 - Detect duplicate records

const fs = require("fs");
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter CSV file path: ", (filePath) => {
    try {
        const data = fs.readFileSync(filePath, "utf-8");

        const lines = data.split("\n");

        const unique = new Set();
        const duplicates = new Set();

        lines.forEach(line => {
            if (unique.has(line)) {
                duplicates.add(line);
            } else {
                unique.add(line);
            }
        });

        console.log("\nDuplicate Records:");
        if (duplicates.size === 0) {
            console.log("No duplicates found.");
        } else {
            duplicates.forEach(dup => console.log(dup));
        }

    } catch (err) {
        console.log("Error:", err.message);
    }

    rl.close();
});
